### DISCUSSION TITLE
How to call event wire:loading in JavaScript?

### DISCUSSION THREAD

**Discussion by kresnasatya** [livewire/livewire#5343] [1. Help]
Hi, I have case make a select dropdown with select2. Because of the behaviour of select2, I cannot rely on `wire:model` attribute. So, the solution is using `@this` blade directive like below.

```php
<script>
    $(document).ready(function () {
        $('#application').select2();
        $('#application').on('change', function (e) {
            @this.selectedClient = e.target.value;
            // I want to trigger wire:loading in this script
        });
    });
</script>

<div wire:ignore>
     <select id="application">
         <option value="">Application</option>
              @foreach ($clients as $client)
               <option value="{{ $client['id'] }}">{{ $client['clientId'] }}</option>
                @endforeach
           </select>
</div>

<!-- I would like to trigger this wire:loading event with JavaScript. How to do that in livewire? -->
<div wire:loading.flex wire:target="selectedClient, refreshClientRolesTable, search" class="justify-center my-2">
     <p class="text-blue-500 semibold tracking-wide">Loading...</p>
</div>
```

So far I know that `wire:loading` can be triggered with `wire:click` and `wire:model`. I would like to know how to trigger `wire:loading` in JavaScript. Thanks.

**Comment by kresnasatya** [comment]
This discussion can be alternative way: https://github.com/livewire/livewire/discussions/2355. Because currently documentation doesn't mention how to call `wire:loading` in JS.

**Comment by JeffGepiga** [comment]
have you tried using `@this.set('selectedClient',this.value) ` ?

**Reply by kresnasatya** [reply]
I have tried that and the result still doesn't work. I think the command `@this.selectedClient = e.target.value` is same with `@this.set('selectedClient',this.value)`
